torch/autograd/profiler.py

```diff
@@ -410,6 +433,13 @@ def _cuda_memory_usage(mem_record):
                 else 0
             )
 
+        def _xpu_memory_usage(mem_record):
+            return (
+                mem_record.nbytes()
+                if mem_record.device_type() in [DeviceType.XPU]
```

Comment:
curious why is XPU special case when we have both CUDA and HIP covered in _cuda_memory_usage. Maybe we can name it _accelerator_memory_usage() but adding a small function for _xpu only seems not required.

Reply:
I am just trying not to modify _cuda named code.